ros2service/ros2service/verb/echo.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+from typing import Optional
+from typing import TypeVar
+
+from collections import OrderedDict
+
+import rclpy
+from rclpy.node import Node
+from rclpy.qos import QoSPresetProfiles
+
+from ros2cli.node.strategy import NodeStrategy
+from ros2topic.api import unsigned_int
+from rcl_interfaces.msg import ServiceEventType
+
+from ros2service.api import ServiceNameCompleter, get_service_class
+from ros2service.api import ServiceTypeCompleter
+from ros2service.verb import VerbExtension
+
+from rosidl_runtime_py import message_to_csv, message_to_ordereddict
+from rosidl_runtime_py.utilities import get_message, get_service
+
+import yaml
+
+DEFAULT_TRUNCATE_LENGTH = 128
+MsgType = TypeVar('MsgType')
+
+
+def represent_ordereddict(dumper, data):
+    items = []
+    for k, v in data.items():
+        items.append((dumper.represent_data(k), dumper.represent_data(v)))
+    return yaml.nodes.MappingNode(u'tag:yaml.org,2002:map', items)
+
+
+class EchoVerb(VerbExtension):
+    """Echo a service."""
+
+    def __init__(self):
+        super().__init__()
+        self.no_str = None
+        self.no_arr = None
+        self.csv = None
+        self.flow_style = None
+        self.client = None
+        self.server = None
+        self.truncate_length = None
+        self.include_message_info = None
+        self.srv_module = None
+        self.event_enum = None
+        self.event_msg_type = get_message("rcl_interfaces/msg/ServiceEvent")
+        self.qos_profile = QoSPresetProfiles.get_from_short_key("services_default")
+        self.__yaml_representer_registered = False
+
+    def add_arguments(self, parser, cli_name):
+        arg = parser.add_argument(
+            'service_name',
+            help="Name of the ROS service to echo from (e.g. '/add_two_ints')")
+        arg.completer = ServiceNameCompleter(
+            include_hidden_services_key='include_hidden_services')
+        arg = parser.add_argument(
+            'service_type', nargs='?',
+            help="Type of the ROS service (e.g. 'std_srvs/srv/Empty')")
+        arg.completer = ServiceTypeCompleter(
+            service_name_key='service_name')
```

Comment:
Looks like this can fit on one line
```suggestion
        arg.completer = ServiceTypeCompleter(service_name_key='service_name')
```